plugins/modules/ipaautomountlocation.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Authors:
+#   Chris Procter <cprocter@redhat.com>
+#
+# Copyright (C) 2019 Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.0",
+    "supported_by": "community",
+    "status": ["preview"],
+}
+
+
+DOCUMENTATION = '''
+---
+module: ipa_automountlocation
+author: chris procter
+short_description: Manage FreeIPA autommount locations
+description:
+- Add and delete an IPA automount location
+options:
+  ipaadmin_principal:
+    description: The admin principal
+    default: admin
+  ipaadmin_password:
+    description: The admin password
+    required: false
+  name:
+    description: The automount location to be managed
+    required: true
+    aliases: ["cn","location"]
+  state:
+    description: State to ensure
+    required: false
+    default: present
+    choices: ["present", "absent"]
+'''
+
+EXAMPLES = '''
+  - name: ensure a automount location named DMZ exists
+    ipaautomountlocation:
+      ipaadmin_password: password01
+      name: DMZ
+      state: present
+
+  - name: remove an automount location named DMZ if it exists
+    ipaautomountlocation:
+      ipaadmin_password: password01
+      name: DMZ
+      state: absent
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.ansible_freeipa_module import FreeIPABaseModule
+
+
+class AutomountLocation(FreeIPABaseModule):
+
+    ipa_param_mapping = {}
+
+    def get_location(self, location):
+        response = self.api_command("automountlocation_find",
+                                    location,
+                                    args={"cn": location})
+
+        if response["count"] == 1:
+            location = response["result"][0]
+        else:
+            location = None
+        return location
+
```

Comment:
When ensuring state is absent, we allow a list of objects in other modules. To add such support, add a `check_ipa_params` implementation:
```python
    def check_ipa_params(self):
        if len(self.ipa_params.name) == 0:
            self.fail_json(msg="At least one location must be provided.")
```

Reply:
This method is still needed, as one might set an empty list as in `name: []`.